### DISCUSSION TITLE
Layers + node_modules issue

### DISCUSSION THREAD

**Discussion by illusive-ch** [hammerstonedev/sidecar#118] [Q&A]
I might be doing this wrong however it seems when I use the @sparticuz/chrome-aws-lambda or any layer for that matter it negates everything that is inside my node_modules directory.

For example, I have these 2 lines of code
```js
const chromium = require("@sparticuz/chrome-aws-lambda");
const { addExtra } = require('puppeteer-extra')
```
```php
public function layers()
    {
        return [
            'arn:aws:lambda:us-west-1:764866452798:layer:chrome-aws-lambda:31'
        ];
    }
```
The lambda function can find "chrome-aws-lambda" using the layer but gives me this error when trying to find "puppeteer-extra"

If I get rid of the layer it will find the "puppeteer-extra" 

Any thoughts on this or am I doing it completely wrong?

**Comment by Akecel** [comment]
Hi there !
I don't know if you found a solution to your problem, but I managed to solve my problem which was similar to yours, so I'm sharing it here !

You will find the procedure to use here in the documentation : https://hammerstone.dev/sidecar/docs/main/functions/handlers-and-packages#separate-modules

But to keep things simple, you'll need to create a `package.json` in your subfolder and add your dependencies (`pupperter-extra`, for example), thus generating the `npm-modules` directory. You can technically push this directly into your lambda function, but it'll be pretty heavy, so the best thing to do is to use `ncc` to manage a dist folder with a single file containing your code and all your dependencies, so that by slightly modifying your SideCar config, you can get your lambda function working ;)
